Run tests separately for different packages in CI

delucis opened the pull request:
## Changes

- Adds separate scripts for testing `astro` and all other packages
- Updates the CI workflow to run each of these as a separate job
- The aim is to see if we can save some time by not running all tests in one big job

## Testing

<!-- How was this change tested? -->
<!-- DON'T DELETE THIS SECTION! If no tests added, explain why. -->

## Docs

<!-- Could this affect a user’s behavior? We probably need to update docs! -->
<!-- If docs will be needed or you’re not sure, uncomment the next line: -->
<!-- /cc @withastro/maintainers-docs for feedback! -->

<!-- DON'T DELETE THIS SECTION! If no docs added, explain why.-->
<!-- https://github.com/withastro/docs -->

ematipico commented:
You might want to add a new fake test, otherwise turborepo won't run them because they're cached

ematipico commented on package.json:
```diff
@@ -21,6 +21,8 @@
     "format:imports": "biome check --formatter-enabled=false --write",
     "format:imports:ci": "biome ci --formatter-enabled=false",
     "test": "turbo run test --concurrency=1 --filter=astro --filter=create-astro --filter=\"@astrojs/*\"",
```
Should we update this command to run `test:astro` and `test:integrations` now?

delucis replied:
Done in ec0a82af400188aa4056569783c8c5b09672df2b 👍

ematipico commented on .github/workflows/ci.yml:
```diff
@@ -103,19 +103,26 @@ jobs:
+          - { name: integrations, script: 'pnpm run test:integrations' }
+        exclude:
           - os: macos-14
-            NODE_VERSION: 22
+            NODE_VERSION: 18
+          - os: macos-14
+            NODE_VERSION: 20
+          - os: windows-2025
+            NODE_VERSION: 18
           - os: windows-2025
-            NODE_VERSION: 22
+            NODE_VERSION: 20
```
Could you explain how this exclusion works?

ematipico reviewed (approved):
Looks good. Letf a couple of comments

delucis commented:
@ematipico You asked on Discord whether this split still preserves the behaviour that we re-run integrations and adapter tests when `astro` changes, so I double checked locally and at least there it behaves as I expected.

I did the following:

1. `pnpm build` to build the monorepo
2. `pnpm build` again to confirm I got a cached run
3. `pnpm test:integrations` to run tests for everything except `astro`
4. `pnpm test:integrations` again to check I got a “full turbo” cached run
5. Modified a file in `astro` (I just added a `console.log`)
6. `pnpm build` to build my changes — it triggered a rebuild as expected
7. `pnpm test:integrations` to see if tests re-ran — they did ✅
